File: bank_simulator/account.cpp
/*
Implementation file for account class
ALwalid Kiawu, 7/7/2026, 9:12PM
*/

#include "account.hpp"
#include "client.hpp"
#include <cstring>
using namespace std;

//constructor
Account::Account(Client client, int Pin){
    client_name = client.getName();
    account_number = 0;
    PIN = Pin;
    balance = 0;
    statement = "";

    //hash to generate account number
    string s_num = "";
    
    for(unsigned char x : client.getName()){
        if(s_num.size() < 7)break;
            s_num += to_string(int(tolower(x)));
            
        }
    
    while(s_num.size() < 7){
        s_num.push_back('0');
    }  
    
    this->account_number = stoi(s_num.substr(0, 7));

    }

    //Getters
int Account::getAccountNumber(){
    return account_number;
}

string Account::getClientName(){
    return client_name;
}

double Account::getBalance(){
    return balance;
}

int Account::getPIN(){
    return PIN;
}

string Account::getStatement(){
    return statement;
}

void Account::displayAccountInfo(){
    cout << "Account Number: " << account_number << endl;
    cout << "Balance: $" << balance << endl;
}

//deposit and withdrawal
void Account::deposit(double amount){
    balance += amount;
    statement +="\n";
    statement +="You deposited " + to_string(amount) + " to your account";
}

void Account::withdraw(double amount){
    balance -= amount;
    statement += "\n";
    statement += "You withdrew " + to_string(amount) + " from your account";
}

File: bank_simulator2.0/models/account.py
"""
ALwalid Kiawu, 8/2/2026, 11:11AM
Account Class
"""


class Account():

    def __init__(self, client, PIN):
        name = client.get_name()
        self.PIN = PIN
        self.balance = 0
        self.statement = ""

        #hash funtion to generate account number
        str_acc_num = ""
        for letter in name:
            if len(str_acc_num) >= 7:
                break
            str_acc_num += str(ord(letter.lower()))

        while len(str_acc_num) < 7:
            str_acc_num += '0'

        self.account_number = int(str_acc_num[:7])

    def display_info(self):
        print(f"Account Number: {self.account_number}")
        print(f"Balance: ${self.balance}")

    def deposit(self, amount:float):
        self.balance += amount
        self.statement += f"You deposited ${amount:.2f} to your account\n"

    def withdraw(self, amount:float):
        self.balance -= amount
        self.statement += f"You withdrew ${amount:.2f} from your account\n"

    def get_statement(self):
        return self.statement

    def get_account_num(self):
        return self.account_number

    def get_balance(self):
        return self.balance

    def get_PIN(self):
        return self.PIN



    









    